lib/benchee/configuration.ex

```diff
@@ -12,53 +12,52 @@ defmodule Benchee.Configuration do
-    parallel:             1,
-    time:                 5,
-    warmup:               2,
-    formatters:           [Console],
-    print: %{
-      benchmarking:       true,
-      configuration:      true,
-      fast_warning:       true
-    },
-    inputs:               nil,
-    save:                 false,
-    load:                 false,
-    # formatters should end up here but known once are still picked up at
-    # the top level for now
-    formatter_options: %{
-      console: %{
-        comparison:          true,
-        extended_statistics: false
-      }
-    },
-    unit_scaling:         :best,
-    # If you/your plugin/whatever needs it your data can go here
-    assigns:              %{},
-    before_each:          nil,
-    after_each:           nil,
-    before_scenario:      nil,
-    after_scenario:       nil
-  ]
+  defstruct parallel: 1,
+            time: 5,
+            warmup: 2,
+            dry_run: false,
+            formatters: [Console],
+            print: %{
+              benchmarking: true,
+              configuration: true,
+              fast_warning: true
+            },
+            inputs: nil,
+            save: false,
+            load: false,
+            # formatters should end up here but known once are still picked up at
+            # the top level for now
+            formatter_options: %{
+              console: %{
+                comparison: true,
+                extended_statistics: false
+              }
+            },
+            unit_scaling: :best,
+            # If you/your plugin/whatever needs it your data can go here
+            assigns: %{},
+            before_each: nil,
+            after_each: nil,
+            before_scenario: nil,
+            after_scenario: nil
 
   @type t :: %__MODULE__{
-    parallel:          integer,
-    time:              number,
-    warmup:            number,
-    formatters:        [((Suite.t) -> Suite.t)],
-    print:             map,
-    inputs:            %{Suite.key => any} | nil,
-    save:              map | false,
-    load:              String.t | [String.t] | false,
-    formatter_options: map,
-    unit_scaling:      Scale.scaling_strategy,
-    assigns:           map,
-    before_each:       fun | nil,
-    after_each:        fun | nil,
-    before_scenario:   fun | nil,
-    after_scenario:    fun | nil
-  }
+          parallel: integer,
+          time: number,
+          warmup: number,
+          formatters: [(Suite.t() -> Suite.t())],
+          print: map,
+          inputs: %{Suite.key() => any} | nil,
+          save: map | false,
+          load: String.t() | [String.t()] | false,
+          formatter_options: map,
+          unit_scaling: Scale.scaling_strategy(),
+          assigns: map,
+          before_each: fun | nil,
+          after_each: fun | nil,
+          before_scenario: fun | nil,
+          after_scenario: fun | nil
+        }
 
   @type user_configuration :: map | keyword
   @time_keys [:time, :warmup]
```

Comment:
`dry_run` is missing in the docs for `init` below here... it's basically a copy from the README I know... haven't foudn a good way to make this work. It definitely belongs into the README imo but would be weird for the main function not to have the docs and sharing the docs seems hard. Maybe we start to just give a run down of the most important options in the README and link to the docs for everything else?